src/azure-cli-core/azure/cli/core/_config.py

```diff
@@ -8,83 +8,32 @@
 
 from azure.cli.core._environment import get_config_dir
 
+from knack.config import get_config_parser, CLIConfig
+
 GLOBAL_CONFIG_DIR = get_config_dir()
 CONFIG_FILE_NAME = 'config'
 GLOBAL_CONFIG_PATH = os.path.join(GLOBAL_CONFIG_DIR, CONFIG_FILE_NAME)
 ENV_VAR_PREFIX = 'AZURE_'
 DEFAULTS_SECTION = 'defaults'
 
-_UNSET = object()
-_ENV_VAR_FORMAT = ENV_VAR_PREFIX + '{section}_{option}'
-
-
-def get_config_parser():
-    import sys
-
-    python_version = sys.version_info.major
-    return configparser.ConfigParser() if python_version == 3 else configparser.SafeConfigParser()
+class AzCliConfig(CLIConfig):
 
+    def set_global_config(config):
+        if not os.path.isdir(GLOBAL_CONFIG_DIR):
+            os.makedirs(GLOBAL_CONFIG_DIR)
+        with open(GLOBAL_CONFIG_PATH, 'w') as configfile:
+            config.write(configfile)
+        os.chmod(GLOBAL_CONFIG_PATH, stat.S_IRUSR | stat.S_IWUSR)
+        # reload az_config
+        az_config.config_parser.read(GLOBAL_CONFIG_PATH)
```

Comment:
Okay, wasn't sure if the knack config was "global" so I kept these around initially.